src/data/set/lattice.lean

```diff
@@ -1573,19 +1573,19 @@ lemma subset_right_of_subset_union (h : s ⊆ t ∪ u) (hab : disjoint s t) : s
 hab.left_le_of_le_sup_left h
 
 lemma preimage {α β} (f : α → β) {s t : set β} (h : disjoint s t) : disjoint (f ⁻¹' s) (f ⁻¹' t) :=
-λ x hx, h hx
+disjoint_iff_inf_le.mpr $ λ x hx, h.le_bot hx
 
 end disjoint
 
 namespace set
 
-protected theorem disjoint_iff : disjoint s t ↔ s ∩ t ⊆ ∅ := iff.rfl
+protected theorem disjoint_iff : disjoint s t ↔ s ∩ t ⊆ ∅ := disjoint_iff_inf_le
 
 theorem disjoint_iff_inter_eq_empty : disjoint s t ↔ s ∩ t = ∅ :=
 disjoint_iff
 
 lemma not_disjoint_iff : ¬disjoint s t ↔ ∃ x, x ∈ s ∧ x ∈ t :=
-not_forall.trans $ exists_congr $ λ x, not_not
+disjoint_iff_inf_le.not.trans $ not_forall.trans $ exists_congr $ λ x, not_not
```

Comment:
```suggestion
by simp [disjoint]
```
should work

Reply:
I figured this was better without `simp` doing something mysterious behind the scenes, and using `simp only` gives a longer and slower proof. I did replace `disjoint_iff_inf_le` with `set.disjoint_iff` though.